src/ninja.cc

```diff
@@ -956,19 +985,51 @@ int NinjaMain::ToolCompilationDatabase(const Options* options, int argc,
 
   optind = 1;
   int opt;
-  while ((opt = getopt(argc, argv, const_cast<char*>("hx"))) != -1) {
+  const option long_options[] = {
+      { "targets", required_argument, nullptr, 't' },
+      { nullptr, 0, nullptr, 0 }
+  };
+
+  std::vector<Edge*> edges_to_process = state_.edges_;
+  while ((opt = getopt_long(argc, argv, "hx:", long_options, nullptr)) != -1) {
     switch(opt) {
       case 'x':
         eval_mode = ECM_EXPAND_RSPFILE;
         break;
 
+      case 't': {
+        std::istringstream iss(optarg);
+        std::string target;
+        std::vector<Edge*> user_interested_edges;
+        DependentEdgeCollector deps_edge_collector;
+
+        while (std::getline(iss, target, ',')) {
```

Comment:
Use `SplitStringPiece()` from `string_piece_util.h` instead for simpler and smaller generated code. Use `std::` prefixes for all standard types, as we are trying to move away from `using namespace std`. E.g.:

  ```
  for (StringPiece target_piece : SplitStringPiece(target, ',')) {
    std::string err;
    std::string target = target_piece.AsString();
    Node* node = CollectTarget(target.c_str(), &err);
    ...
  }